### DISCUSSION TITLE
pci-e card workaround for autoflash

### DISCUSSION THREAD

**Discussion by brcisna** [danielewood/sierra-wireless-modems#85] [General]
Hello All,

New to Sierra Wirelss cards.

Purchased this pci-e to M2-adapter for em7455 Sierra Wireless card

https://www.aliexpress.com/item/33040315721.html?src=google&src=google&albch=shopping&acnt=708-803-3821&slnk=&plac=&mtctp=&albbt=Google_7_shopping&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=9765122064&albag=101238896618&trgt=892975210462&crea=en33040315721&netw=u&device=c&albpg=892975210462&albpd=en33040315721&gclid=CjwKCAjwmqKJBhAWEiwAMvGt6NL9FRqYEgEKVi7SnSkQTybQ4MbnbQINfBo__AjupA1qKkPrLk9CTBoC8VYQAvD_BwE&gclsrc=aw.ds&aff_fcid=b978e093192141808af3bbadb530ff4c-1630110695677-06349-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=b978e093192141808af3bbadb530ff4c-1630110695677-06349-UneMJZVf&terminal_id=f712168c05de49e28338fa965bc07a4c

Along with a Dell branded em7455 Sierra Wireless card

is there any way of modding the autiinstall script to flash firmware with the setup?

Thank You

**Comment by danielewood** [comment]
You don't need to modify anything. The script will by default convert the dell to a generic Sierra in the process of updating it.

**Comment by brcisna** [comment]
OK Daniel,,
Thank You for this info.
 I know the instructions says Warning:  "Has to be the USB external
adapter." I thought the way the script looked it tried to parse USB
device/modem?..
Maybe even on the pci-e adapter to M2 slot,it ends up being detected as a
usb? Never messed with these before,
 As soon as modem arrives ill give the autoinstall a try.

Thank You

**Reply by danielewood** [reply]
In this case, the PCIe adapter might not even route PCIe signals to the modem, at all. The connector on the back for plugging into your motherboard's USB headed will handle everything. If you run into issues, I suggest taping over the PCIe connector with kapton tape or unplugging it from the PCIe slot while flashing.

**Comment by brcisna** [comment]
Ok,,,Daniel,

Thanks once again for this info.
I couldn't even figure out why this adapter had the USB connector on it,,in
the first place.. I think for external drive maybe? It did in fact bundle
with the 7 pin? USB cable..Ill give your suggestion a shot,,and lay down a
report .

Thank You

**Reply by danielewood** [reply]
There is a slight misunderstanding, the PCIe slot itself will not be used for anything other than a physical support structure to hold the card in place. 

All communications with the modem will be 💯 over that 7-pin USB header(of which, only 4 pins will be used). You do not even need to plug the card into a PCIe slot at all.

**Comment by brcisna** [comment]
Ok,,,Daniel,,,

Thanks again for information. Am showing my stupidity on this setup, it
goes without saying.
So,, the usb cable is imperative to be used on this setup.
Curious to see how the autoinstall script goes.
My em7455  Sierra Wireless, card, shipping has been 'stuck' for 8 days now
after arriving stateside,,,at same local.  via AliExpress.

Thanks again

**Comment by brcisna** [comment]
Hi Daniel,

Tried the pci-e card that i purchased, Plugged into an open pci-e slot
plugged usb cable into motherboard. Booted the Ubuntu Live 18.04. Nothing
detected in lspci or lsusb. I wonder if the card needs the actual modem in
it to be recognized as anything? The modem is dead in transit at inbound
postal from China.
Was very skeptcal in getting this card,,but thought it may be slightly more
robust than just a usb dongle.

Thanks

**Reply by danielewood** [reply]
As expected. All the card does is physical adaptation. There are no brains on the card. 

You've done the same thing as insert a microsd to SD card adapter with no microsd card inserted.

**Comment by brcisna** [comment]
Thanks for the tip!
Makes sense.  Hopefully the em7455 works like a champ in this setup.

**Comment by brcisna** [comment]
Hi daniel,

I finally received a NEW em7455 Sierra Wireless modem to plug into the adapter i linked above.
Using ONLY the supplied usb cable plugging into an open usb header on mobo, there is still nothing detected as a new USB device running lsusb.
So, plugged the adapter into an open pci-e slot , along with having the usb cable connected to usb header on mobo, booted into the Ubuntu live CD and still no usb device listed for this card,as well as no new pci device showing in lspci. The adapter led does at least come on on power up.
Have never seen one post if anyone has used this particular adapter for any uses at all? I thought this may be a slightly more clean/robust setup than the external usb plastic enclosuer setup. Am wondering if the (other) slot,which is actually listed as 'slot 1' on the spec sheet has to be in populated  for the second slot em7455 to function?
Only have one pc with an open usb header to test on here in the decommissioned pc junk pile.

Thanks

**Reply by danielewood** [reply]
lsusb is the only place you should see any changes. The PCIe lanes will be reserved for a WiFi card exclusively. 

It is possible you have a dead modem, it is also possible there is some other underlying problem with your configuration. 

I was wiring a PCIe to m2 WiFi adapter last week, it only used the two data pins on the USB header(only those two pins had any PCB traces). Power was supplied by the PCIe connector. 

The fact that you have blinky lights bodes well. I assume your current configuration is the card in the PCIe slot AND the USB header plugged in. If that is the case and you still have nothing useful in lsusb, I'm really not sure where you go from here.

**Comment by brcisna** [comment]
Daniel

Got the adapter card into a newer pc/motherboard 
A DELL Corp 413c:81b6 is now showing in lsusb
I used the latest autoflash-7455.sh zip
Failed initially with no curl found. Installed curl.
Looks like the modem is starting to be flashed.but hangs forever at :
"Waiting for modem in boothold mode..."

using Ubuntu 20.04  
Previously i seen a failed may need to install UUID.Tiny.pm module,,,but it does proceed on past this to the ,,what look like starting to autoflash.
Until you have one of these do a successful flash you dont know what you should be seeing in the terminal,,i guess.

Thanks agai
![IMG_20210922_132822588](https://user-images.githubusercontent.com/47399290/134402534-5357035f-4a7c-428f-804e-cad1487b60f7.jpg)
n

**Reply by brcisna** [reply]
Success!

Got the EM7455 to successfully flash!
Problem was i had forgot to allow this machine to do downloads as i was doing this at work. Couldnt figure out why the repo updates kept failing .it appeard the repos were not 'signed' After bypassing the firewall then everything took off.
I did have to change one line in the autoinstall-7455.sh about changing to the particular port,,the em7455 was on, ,which mine ended up being /dev/ttyUSB2.
Someone else had posted this workaround several months ago here.
The little blinky led did go out on the adapter card a couple times while modem rebooted and had fingers crossed blue blinky would come back to life which it did!
Question:
Is there any chance this will work with an active sim through NetworkManager GUI using only the APN setting for the particular carrier?
![IMG_20210923_090819608](https://user-images.githubusercontent.com/47399290/134524257-c76cf6c2-f04a-4505-89c3-6adfce63d270.jpg)

**Comment by danielewood** [comment]
> Is there any chance this will work with an active sim through NetworkManager GUI using only the APN setting for the particular carrier?

Yes, it should work fine. I have had Ubuntu LiveCDs just connect right up with nothing more than specifying the APN and these modems with an active SIM.
